docs/amber_script.md

```diff
@@ -458,6 +557,37 @@ The following commands are all specified within the `PIPELINE` command.
   PATCH_CONTROL_POINTS {control_points}
 ```
 
+Ray tracing pipelines do not attach shaders directly like compute or graphics pipelines.
+Ray tracing pipelines organize shaders into shader groups in one of four ways
+depending on shader types used:
+
+```groovy
+  # Four possible shader group definitions
+  SHADER_GROUP {group_name_1} {ray_generation_shader_name}
+  SHADER_GROUP {group_name_2} {miss_shader_name}
+  SHADER_GROUP {group_name_3} {call_shader_name}
+  SHADER_GROUP {group_name_4} {any_hit_shader_name} {closest_hit_shader_name} {intersection_shader_name}
+```
+
+Each group name must be unique within a pipeline. The same shader can be used within one or more
+shader groups. The shader group order is important, further commands as shader code might refer
+them directly. With the shader groups defined, they are then added into shader binding tables:
+
+```groovy
+  # Create shader binding tables and set shader groups into it
+  SHADER_BINDING_TABLE {sbt_name}
+    {group_name_1}
+    [ | {group_name_n}]
+  END
+```
```

Comment:
How are non-existing/null shaders (e.g. null AHS) specified here?

Reply:
> How can max payload and hit attribute sizes be specified?

I am not sure what do you mean. These values are fully inside shader interfaces and do not need any regulation through scripting language.

> Are pipeline libraries (planned to be) supported?

Not sure yet.

> How are non-existing/null shaders (e.g. null AHS) specified here?

I should update documentation to:

SHADER_GROUP {group_name_4} [any_hit_shader_name] [closest_hit_shader_name] [intersection_shader_name]

i.e. you can skip unused shaders, actually code does not enforce shader order two only things checked: (1) is that code does not mix general shaders with hit shaders; (2) there are no two same types within group (i.e. group does not include two+ intersection shaders or two+ anyhit shaders or two+ closesthit shaders).